gradle/defaults-java.gradle

```diff
@@ -60,3 +60,14 @@ allprojects {
     }
   }
 }
+
+configure(project(":lucene:core")) {
```

Comment:
So if we want this in before JDK16 becomes the minimum platform we'd need to use a multi-release jar again, right?

Reply:
We can't release that anyways. Earliest ist jdk 17, if not 18. The API is not stable at all.

Reply:
Ok, just wondering!

Reply:
The main reason for this PR is testing and feedback to JDK team.
Jenkins tests it and we can feedback. 4 issues already reported.

Reply:
To come back to the MR-JAR: Yes we may need it, but it can be solved without, too.

The IndexInputs are completely separate (so they can life side-by side). The changes to MMapDircetory are mainly a change from int to long for the chunk size (this is why I also not yet removed the methods to query unmapping status, thy would always return true for JDK>xxxx.

When we would have this productive, MMapDirectory would just return one of the IndexInputs that fits your JDK. We may only need to move the code that maps a file to ByteBuffer or MemorySegment to a separate class or maybe the IndexInput class. Then we have 2 classes staying next to each other, one for JDK 11 and one for JDK 16/17/18+, MMapDircetory just chooses the correct one. Then we won't need a MR JAR.

With compilation we can use `--release` and of course a newer Gradle version. Like Elasticsearch, people who want to compile Lucene/Solr would need JDK 17, but the code is working for JDK 11+. Elasticsearch does exactly the same: It is tested and compiled for Java 8, but to build it you need JDK 15.